### DISCUSSION TITLE
urql/@vue  How use authentication in Vue SSR

### DISCUSSION THREAD

**Discussion by snaplesec** [urql-graphql/urql#1463] [Q&A] [answered]
How do I use the auth token from the client side of ssr and how to make sure that it's added to the server side.
Also where is the authExchange in the list?  Does it go before or after ssr?

**Comment by kitten** [comment, marked as answer]
There's some details on this here, if you have a larger auth need: https://formidable.com/open-source/urql/docs/advanced/authentication/

> where is the authExchange in the list? Does it go before or after ssr?

It goes after SSR since it's an asynchronous exchange, i.e. it may do auth calls, while the `ssrExchange` and `cacheExchange` are synchronous, i.e. caching, exchanges, so you'll have something like `1. dedup, 2. ssr, 3. cache, 4. auth, 5. fetch`

If you need to cater for SSR then you need to find a different way to get your API credentials for `getAuth`. This will very much depend on your authentication method and how you've set up your app.

**Reply by snaplesec** [reply]
I did see that.  Can you look at my questions in this ticket?  #1465

Thanks,
Steve

**Reply by snaplesec** [reply]
I am using express-jwt on the server, and I get a token back from login.
